### ISSUE TITLE
Split Foundatio.Extensions.Hosting in two

### ISSUE THREAD

**Issue by pinkfloydx33** [FoundatioFx/Foundatio#279]
Currently, `Foundatio.Extensions.Hosting` has a `FrameworkReference` to `Microsoft.AspNetCore.App` in order to accommodate the startup filters and middleware. However, this package isn't just limited to ASP.NET. In fact, we use it inside of a Worker Service (no web component) to leverage the `HostedJobService` and friends. I noticed today while examining the outputs of a self-contained publish that the ASPNET shared framework assemblies were all being included. This is increasing application publish size (and the size of our docker images) and is entirely unnecessary for non-ASP applications.

I suggest splitting the package into two -- one for the hosted service infrastructure and the other for the ASP.NET-specific components.

**Comment by ejsmith** [comment]
Hmm... not opposed to breaking it up, but I'd prefer not to. Our job services still use web just so they can serve health and readiness probes. How much is it adding to the docker image?

**Comment by ejsmith** [comment]
Yeah, what you are saying makes sense. Not sure what we would call the 2 different packages. Ideas? Can you create a PR for the changes?

**Comment by pinkfloydx33** [comment]
Taking Serilog as an example, they call them `Serilog.Extensions.Hosting` and `Serilog.AspNetCore`. The latter is exactly like what I'm proposing with basically Middleware and AppBuilder/ServiceCollection extensions and a reference to/dependency on the former. 

So perhaps:

- `Foundatio.Extensions.Hosting`
- `Foundatio.AspNetCore`

I'm having oral surgery tomorrow. Not sure how willing I'll be to do anything *immediately*, but certainly something I can contribute to.

**Comment by niemyjski** [comment]
We'd be willing to accept a pr for this
